pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1262,6 +1262,35 @@ func (c *Cluster) startMon(m *monConfig, schedule *MonScheduleInfo) error {
 		logger.Debugf("adding host path volume source to mon deployment %s", d.Name)
 	}
 
+	// Expand mon PVC if storage request for mon has increased in cephcluster crd
+	if pvcExists {
+		pvc, err := c.makeDeploymentPVC(m, false)
+		if err != nil {
+			return errors.Wrapf(err, "failed to make mon %s pvc", d.Name)
+		}
+
+		existingPvc, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.Namespace).Get(ctx, m.ResourceName, metav1.GetOptions{})
```

Comment:
The OSDs handle increasing the PVC size with a helper `updatePVCIfChanged()`, how about refactoring that to also be used here for the mons?

Reply:
`updatePVCIfChanged()` is a method on `Cluster` for osd package `Cluster` is different than `Cluster` in the mon package. We can not directly call this method from osd package as we won't have a `Cluster` object of the osd package. If we want to refactor this we have to do it in such a way that we move this method to the common place and pass the required values as an argument.

So the question is where should we move this method?

As of now, I have just copied the same method in the mon package itself and did the required tweaks.

Reply:
You can put this new function in the `opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"` package.

Reply:
Done